urlf91495fd27_Rembau_complie_tgz-pJ8-matcher_MatchJ8/src/regexpToNFA/Syntax.java

```diff
@@ -0,0 +1,136 @@
+package regexpToNFA;
+
+/**
+ * 语法中的一些关键字
+ * @author Administrator
+ *
+ */
+public class Syntax {
+	/**
+	 * 闭包
+	 */
+	public static char closure='*';
+	/**
+	 * 正闭包
+	 */
+	public static char positiveClosure='+';
+	/**
+	 * 或
+	 */
+	public static char or='|';
+	public static char transMean='\\';
+	public static char[] keyChar=new char[]{'*','+','(',')','|','(',')','?'};  //关键字
+	public static char includeLeft='[';
+	public static char includeRight=']';
+	
+	public static char lastChar;
+	public static char nextChar;
+	public static String noPlace[]=new String[]{"\\b","\\c"};   //不占位转移符号  \\b单词开始  \\d单词结束
+	public static char keyCharForS[]=new char[]{'-'};  //中括号里面的关键字
+	
+	public static boolean isNoPlace(String tran){
+		for(String str:noPlace){
+			if(tran.equals(str)){
+				return true;
+			}
+		}
+		return false;
+	}
+	public static boolean isIncludeLeft(char ch){
+		return includeLeft==ch;
+	}
+	public static boolean isIncludeRight(char ch){
+		return includeRight==ch;
+	}
+	/**
+	 * 判断是否是 '*','+','(',')','|','(',')' 中的一个
+	 * @param str
+	 * @return
+	 */
+	public static boolean isKeyWord(char str){
+		for(int i=0;i<keyChar.length;i++){
+			if(keyChar[i]==str){
+				return true;
+			}
+		}
+		return false;
+	}
+	/**
+	 * 判断是否是转义字符\
+	 * @param str
+	 * @return
+	 */
+	public static boolean isThransMean(char str){
+		if(transMean==str){
+			return true;
+		}
+		return false;
+	}
+	/**
+	 * 判断是否匹配
+	 * @param tran 转换字符 可以是char或者是\w关键字
+	 * @param in 输入的char
+	 * @return
+	 */
+	public static boolean isMatch(String tran,char in){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.